tests/components/prometheus/test_init.py

```diff
@@ -639,6 +640,351 @@ async def test_lock(hass, hass_client):
     )
 
 
+async def test_renaming_entity_name(hass, hass_client):
+    """Test renaming entity name."""
+    assert await async_setup_component(
+        hass,
+        "conversation",
+        {},
+    )
+    client = await setup_prometheus_client(hass, hass_client, "")
+
+    assert await async_setup_component(
+        hass, sensor.DOMAIN, {"sensor": [{"platform": "demo"}]}
+    )
+
+    await hass.async_block_till_done()
+    body = await generate_latest_metrics(client)
+
+    assert (
+        'sensor_temperature_celsius{domain="sensor",'
+        'entity="sensor.outside_temperature",'
+        'friendly_name="Outside Temperature"} 15.6' in body
+    )
+
+    assert (
+        'entity_available{domain="sensor",'
+        'entity="sensor.outside_temperature",'
+        'friendly_name="Outside Temperature"} 1.0' in body
+    )
+
+    assert (
+        'sensor_humidity_percent{domain="sensor",'
+        'entity="sensor.outside_humidity",'
+        'friendly_name="Outside Humidity"} 54.0' in body
+    )
+
+    assert (
+        'entity_available{domain="sensor",'
+        'entity="sensor.outside_humidity",'
+        'friendly_name="Outside Humidity"} 1.0' in body
+    )
+
+    registry = entity_registry.async_get(hass)
+    assert "sensor.outside_temperature" in registry.entities
+    registry.async_update_entity(
+        entity_id="sensor.outside_temperature",
+        name="Inside Temperature",
+    )
+
+    await hass.async_block_till_done()
+    body = await generate_latest_metrics(client)
+
+    # Check if old metrics deleted
+    assert (
+        'sensor_temperature_celsius{domain="sensor",'
+        'entity="sensor.outside_temperature",'
+        'friendly_name="Outside Temperature"} 15.6' not in body
+    )
+
+    assert (
+        'entity_available{domain="sensor",'
+        'entity="sensor.outside_temperature",'
+        'friendly_name="Outside Temperature"} 1.0' not in body
+    )
```

Comment:
I would suggest a broader check here, perhaps just `assert 'friendly_name="Outside Temperature"' not in '\n'.join(body)`?

This applies to all other similar cases below.

Reply:
Thanks, this seems to be the better solution! Added it in the new commit.